kedro/framework/project/__init__.py

```diff
@@ -216,10 +216,28 @@ def configure(self, pipelines_module: str | None = None) -> None:
 class _ProjectLogging(UserDict):
     def __init__(self) -> None:
         """Initialise project logging. The path to logging configuration is given in
-        environment variable KEDRO_LOGGING_CONFIG (defaults to default_logging.yml)."""
-        path = os.environ.get(
-            "KEDRO_LOGGING_CONFIG", Path(__file__).parent / "default_logging.yml"
-        )
+        environment variable KEDRO_LOGGING_CONFIG (defaults to conf/logging.yml)."""
+
+        # Check if a user path is set in the environment variable
+        user_logging_path = os.environ.get("KEDRO_LOGGING_CONFIG")
+
+        # Check if the default logging configuration exists
+        default_logging_path = Path("conf/logging.yml")
+        if not default_logging_path.exists():
+            default_logging_path = Path(__file__).parent / "default_logging.yml"
+
+        # Use the user path if available, otherwise, use the default path
+        if user_logging_path and Path(user_logging_path).exists():
+            path = Path(user_logging_path)
+        else:
+            path = default_logging_path
+            logger = logging.getLogger(__name__)
+            logger.warning(
+                "Using `conf/logging.yml` as logging configuration. You can change this by setting the "
+                "KEDRO_LOGGING_CONFIG environment variable accordingly."
+            )
```

Comment:
I'm on the fence on whether this should be `logger.warning` or `warnings.warn`... From https://docs.python.org/3/howto/logging.html#when-to-use-logging,

> [warnings.warn()](https://docs.python.org/3/library/warnings.html#warnings.warn) in library code if the issue is avoidable and the client application should be modified to eliminate the warning
> 
> A logger’s [warning()](https://docs.python.org/3/library/logging.html#logging.Logger.warning) method if there is nothing the client application can do about the situation, but the event should still be noted

Reply:
I see, then I think then it makes sense to use `warnings.warn` because it lets the users know they can make modifications to avoid this.

Reply:
I don't think this should be a warning at all - could be an info level log. If a user is not bothered about the logging stuff, it might be annoying if this keeps popping up with every `kedro run`. There's nothing technically wrong with just using the default logging settings. 

Additionally, since this warning is emitted before the logging settings are configured, it looks kinda out of place. It should be shown after the `self.configure(..)` step. WDYT?

![Screenshot 2024-05-07 at 11 14 00](https://github.com/kedro-org/kedro/assets/110245118/4cb65d13-e0a1-4b2d-b557-e86536f4589b)

Reply:
> If a user is not bothered about the logging stuff, it might be annoying if this keeps popping up with every kedro run. There's nothing technically wrong with just using the default logging settings.

Good point @ankatiyar, agreed 👍🏼